animations/hero-animations/index.md

```diff
@@ -0,0 +1,670 @@
+
+  Widget build(BuildContext context) {
+    return new SizedBox(
+      width: width,
+      child: new Hero(
+        tag: photo,
+        child: new Material(
+          color: Colors.transparent,
+          child: new InkWell(
+            onTap: onTap,
+            child: new Image.asset(
+              photo,
+              fit: BoxFit.contain,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+Key information:
+
+* An InkWell wraps the hero, making it trivial to add a tap
+  gesture to the both the source and destination heroes.
+* Defining the Material widget with a transparent color
+  enables the image to "pop out" of the background as it
+  flies to its destination.
+* The SizedBox specifies the hero's size at the start and
+  end of the animation.
+* Setting the Image's `fit` property to `BoxFit.contain`,
+  ensures that the image is as large as possible during the
+  transition without changing its aspect ratio.
+
+### HeroAnimation class
+
+The HeroAnimation class creates the source and destination
+PhotoHeroes, and sets up the transition.
+
+Here's the code:
+
+<!-- skip -->
+{% prettify dart %}
+class HeroAnimation extends StatelessWidget {
+  Widget build(BuildContext context) {
+    [[highlight]]timeDilation = 5.0; // 1.0 means normal animation speed.[[/highlight]]
+
+    return new Scaffold(
+      appBar: new AppBar(
+        title: const Text('Basic Hero Animation'),
+      ),
+      body: new Center(
+        [[highlight]]child: new PhotoHero([[/highlight]]
+          photo: 'images/flippers-alpha.png',
+          width: 300.0,
+          [[highlight]]onTap: ()[[/highlight]] {
+            [[highlight]]Navigator.of(context).push(new MaterialPageRoute<Null>([[/highlight]]
+              [[highlight]]builder: (BuildContext context)[[/highlight]] {
+                return new Scaffold(
+                  appBar: new AppBar(
+                    title: const Text('Flippers Page'),
+                  ),
+                  body: new Container(
+                    // The blue background emphasizes that it's a new route.
+                    color: Colors.lightBlueAccent,
+                    padding: const EdgeInsets.all(16.0),
+                    alignment: Alignment.topLeft,
+                    [[highlight]]child: new PhotoHero([[/highlight]]
+                      photo: 'images/flippers-alpha.png',
+                      width: 100.0,
+                      [[highlight]]onTap: ()[[/highlight]] {
+                        [[highlight]]Navigator.of(context).pop();[[/highlight]]
+                      },
+                    ),
+                  ),
+                );
+              }
+            ));
+          },
+        ),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+Key information:
+
+* When the user taps the InkWell containing the source hero,
+  the code creates a new MaterialPageRoute (in `onTap()`) and pushes
+  it to the Navigator’s stack, triggering the animation.
```

Comment:
It would be helpful to remind the reader that this route we've pushed is now the "destination".